src/puppetlabs/puppetdb/nodes.clj

```diff
@@ -0,0 +1,51 @@
+(ns puppetlabs.puppetdb.nodes
+  "Puppet nodes parsing
+
+   Functions that handle conversion of nodes from wire format to
+   internal PuppetDB format, including validation."
+  (:require
+   [schema.core :as s]
+   [clojure.set :as set]
+   [puppetlabs.puppetdb.schema :as pls]
+   [puppetlabs.puppetdb.utils :as utils]
+   [com.rpl.specter :as sp]))
+
+;; SCHEMA
+
+(def expire-wireformat-schema
+  {:facts s/Bool})
+
+(def configure-expiration-wireformat-schema
+  {:certname s/Str
+   :expire expire-wireformat-schema
+   :producer_timestamp (s/maybe pls/Timestamp)})
+
+(def nodes-wireformat-schema
+  {:certname s/Str
+   :deactivated (s/maybe s/Str)
+   :expired (s/maybe pls/Timestamp)
+   :catalog_timestamp (s/maybe pls/Timestamp)
+   :facts_timestamp (s/maybe pls/Timestamp)
+   :report_timestamp (s/maybe pls/Timestamp)
+   :catalog_environment (s/maybe s/Str)
+   :facts_environment (s/maybe s/Str)
+   :report_environment (s/maybe s/Str)
+   :latest_report_status (s/maybe s/Str)
+   :latest_report_hash (s/maybe s/Str)
+   :latest_report_noop (s/maybe s/Bool)
+   :latest_report_noop_pending (s/maybe s/Bool)
+   :cached_catalog_status (s/maybe s/Str)
+   :latest_report_corrective_change (s/maybe s/Bool)
+   :latest_report_job_id (s/maybe s/Str)
+   :expires_facts (s/maybe s/Bool)
+   :expires_facts_updated (s/maybe pls/Timestamp)})
+
+(pls/defn-validated nodes-query->configure-expiration-wire-v1
+  :- [configure-expiration-wireformat-schema]
+  [nodes :- [nodes-wireformat-schema]]
+  (->> nodes
+       (filter #(false? (:expires_facts %)))
```

Comment:
`false?` is only false for the literal value `false`, since `expires_facts` is `nil` this will end up not filtering anything right?

Reply:
```suggestion
       (filter :expires_facts_updated %))
```